include/storage/io_config.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef OSRM_IO_CONFIG_HPP
+#define OSRM_IO_CONFIG_HPP
+
+#include "util/exception.hpp"
+#include "util/log.hpp"
+
+#include <array>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/path.hpp>
+#include <string>
+
+namespace osrm
+{
+namespace storage
+{
+struct IOConfig
+{
+    IOConfig(std::vector<boost::filesystem::path> _required_input_files,
+             std::vector<boost::filesystem::path> _optional_input_files,
+             std::vector<boost::filesystem::path> _output_files)
+        : required_input_files(_required_input_files), optional_input_files(_optional_input_files),
+          output_files(_output_files)
+    {
+    }
+
+    // Infer the base path from the path of the .osrm file
+    void UseDefaultOutputNames(const boost::filesystem::path &base)
+    {
+        // potentially strip off the .osrm (or other) extensions for
+        // determining the base path=
+        std::string path = base.string();
+
+        std::array<std::string, 6> known_extensions{
+            {".osm.bz2", ".osm.pbf", ".osm.xml", ".pbf", ".osm", ".osrm"}};
+        for (auto ext : known_extensions)
+        {
+            const auto pos = path.find(ext);
+            if (pos != std::string::npos)
+            {
+                path.replace(pos, ext.size(), "");
+                break;
+            }
+        }
+
+        base_path = {path};
```

Comment:
I think you can replace the last 15 lines with

    trim_right_if(path, is_any_of("all your extensions"));

http://www.boost.org/doc/libs/1_63_0/doc/html/string_algo/usage.html#idp113215984